SnakeGame/Snake.js

```diff
@@ -0,0 +1,207 @@
+        this.head = [2, 3];
+        // start off as a 3 unit long snake
+        this.squeue.enqueue([2, 1]);
+        this.squeue.enqueue([2, 2]);
+        this.squeue.enqueue([2, 3]);
+        this.appleCount = 0;
+        this.input = "R";
+
+        pageCounter.textContent = "Apples Eaten: " + this.appleCount;
+    }
+
+    move() {
+        let dir = this.direction;
+        if (this.input == "R" && dir == "L"
+            || this.input == "L" && dir == "R"
+            || this.input == "U" && dir == "D"
+            || this.input == "D" && dir == "U") {
+        } else {
+            this.direction = this.input;
+        }
+        dir = this.direction;
+
+        if (dir == "R") {
+            this.squeue.enqueue([this.head[0], this.head[1] + 1]);
+            this.head = [this.head[0], this.head[1] + 1];
+        } else if (dir == "U") {
+            this.squeue.enqueue([this.head[0] - 1, this.head[1]]);
+            this.head = [this.head[0] - 1, this.head[1]];
+        } else if (dir == "L") {
+            this.squeue.enqueue([this.head[0], this.head[1] - 1]);
+            this.head = [this.head[0], this.head[1] - 1];
+        } else if (dir == "D") {
+            this.squeue.enqueue([this.head[0] + 1, this.head[1]]);
+            this.head = [this.head[0] + 1, this.head[1]];
+        }
+        let tail;
+
+        // check if apple is eaten
+        let apple = game.apple;
+        if (this.head[0] == apple[0] && this.head[1] == apple[1]) {
+            tiles[apple[0]][apple[1]].classList.remove("apple");
+            this.appleCount++;
+            this.game.apple = game.newAppleLoc();
+            pageCounter.textContent = "Apples Eaten: " + this.appleCount;
+        } else {
+            tail = this.squeue.dequeue().data;
+        }
+
+        game.tick(this.head, tail);
+    }
+}
+
+class Game {
+    constructor(n) {
+        this.n = n;
+    }
+
+    reset() {
+        this.board = [];
+        for (let i = 0; i < this.n; i++) {
+            let row = [];
+            for (let j = 0; j < this.n; j++) {
+                row.push('-');
+            }
+            this.board.push(row);
+        }
+
+        this.snake = new Snake(this);
+        this.gameState = "NOT_STARTED";
+
+        // reset tiles on page
+        let allTiles = document.getElementsByClassName("snakeBody");
+        while (allTiles.length > 0) {
+            allTiles[0].classList.remove("snakeBody");
+        }
+        let appleTiles = document.getElementsByClassName("apple");
+        while (appleTiles.length > 0) {
+            appleTiles[0].classList.remove("apple");
+        }
+
+        this.apple = this.newAppleLoc();
+        this.debug();
+
+        this.interval = setInterval(function () {
+            game.snake.move();
+        }, 200);
+
+        pageMessages.innerHTML = "Game State: " + game.gameState;
+    }
+
+    newAppleLoc() {
+        let loc = [Math.round(Math.random() * (this.n - 1)), Math.round(Math.random() * (this.n - 1))];
```

Comment:
To generate a number in the range [0, n), we typically use `Math.floor(Math.random()*n)` rather than `Math.round(Math.random() * (n-1))`. I'll need to check, but I believe your version generates a non-uniform distribution. It has a lower chance of choosing one of the edge pieces due to the rounding.